libsql/src/database.rs

```diff
@@ -393,6 +433,21 @@ impl Database {
             _ => unreachable!("no database type set"),
         }
     }
+
+    #[cfg(all(feature = "encryption-at-rest", feature = "replication"))]
+    pub fn set_encryption_key(&mut self, encryption_key: impl AsRef<[u8]>) {
+        match self.db_type {
+            DbType::Sync {
+                db: _,
+                encryption_key: ref mut p,
+            } => *p = Some(bytes::Bytes::from(encryption_key.as_ref().to_owned())),
+            _ => {
+                tracing::warn!(
```

Comment:
I want to say this would be better as a hard error than a log that could be silent if a user doesn't setup the tracing subscriber.